src/DatePair/widget/DateTime.js

```diff
@@ -1,493 +1,527 @@
+                }
+            } else {
+                if (this.dp) {
+                    this.dp.set("clear");
+                }
+                if (this.tp) {
+                    this.tp.set("clear");
+                }
+            }
+
+            this._updateRendering();
+
+        },
+
+        resize: function(box) {
+            logger.debug(this.id + ".resize");
+        },
+
+        _convertToMomentFormatString: function(formatString) {
+            // TIMEPICKER
+            // DATEPICKER
+            // http://amsul.ca/pickadate.js/date/#formatting-rules
+            return formatString
+                .split("m").join("M")
+                .split("d").join("D")
+                .split("yyyy").join("YYYY")
+                .split("g").join("h")
+                .split("G").join("H")
+                .split("i").join("m");
+        },
+
+        _updateRendering: function(callback) {
+            logger.debug(this.id + "._updateRendering");
+
+            if (this._contextObj !== null) {
+                dojoStyle.set(this.domNode, "display", "block");
+            } else {
+                dojoStyle.set(this.domNode, "display", "none");
+            }
+
+            // Important to clear all validations!
+            this._clearValidations();
+
+            this._executeCallback(callback);
+        },
+
+        _executeCallback: function(cb) {
+            if (cb && typeof cb === "function") {
+                cb();
+            }
+        },
+
+        /**
+         * Reset Subscriptions
+         */
+        _resetSubscriptions: function() {
+            this.unsubscribeAll();
+            // attr
+            this.subscribe({
+                guid: this._contextObj.getGuid(), // the guid
+                attr: this.fromDate, // the attributeName
+                callback: lang.hitch(this, function(guid, attr, attrValue) {
+                    this._setDateTimePickerValues(attrValue);
+                })
+            });
+            //object
+            this.subscribe({
+                guid: this._contextObj.getGuid(), // the guid
+                callback: lang.hitch(this, function(guid, attr, attrValue) {
+                    this._setDateTimePickerValues(attrValue);
+                })
+            });
+            this.subscribe({
+                guid: this._contextObj.getGuid(),
+                val: true,
+                callback: lang.hitch(this, function(validations) {
+                    this._handleValidations(validations);
+                })
+            });
+        },
+
+        /**
+         * Handle Validations
+         */
+        _handleValidations: function(validations) {
+            logger.debug(this.id + "._handleValidation");
+            this._clearValidations();
+
+            var validation = validations[0],
+                message = validation.getReasonByAttribute(this.fromDate);
+
+            if (this._readOnly) {
+                validation.removeAttribute(this.fromDate);
```

Comment:
this `validation.removeAttribute` is the only difference I see between your code and mine. Can you please explain why this is beneficial, compared to how I had done it? It looks like this code could still show an incorrect validation based on there being more than one object to validate on the page

Reply:
You don't check if the attribute is readOnly - there's no sense in showing feedback for a field a user cannot change. Additionally you don't check if there's an actual message for the attribute. This will show an empty alert box below the widget if a different attribute in the context has a validation.

`} else if (message) {
    this._addValidation(message);
    validation.removeAttribute(this.fromDate);
}`